synapse/util/async_helpers.py

```diff
@@ -188,13 +188,21 @@ def yieldable_gather_results(func, iter, *args, **kwargs):
     ).addErrback(unwrapFirstError)
 
 
+@attr.s(slots=True)
+class _LinearizerEntry:
+    # The number of things executing.
+    count = attr.ib(type=int)
+    # Deferreds for the things bocked from executing.
+    deferreds = attr.ib(type=collections.OrderedDict)
```

Comment:
Why not `awaitables`. `Deferred`s are `Awaitable`s too, right?

Or is this because the rest of the file refers to them as deferreds?

Reply:
Because it is actually a `Deferred`, not just an `Awaitable` (and since this is internal to the functionality of the `Linearizer` it seems better to be specific).

https://github.com/matrix-org/synapse/blob/8b04ec459df11f65ec1452a183e77eaeead4ddd3/synapse/util/async_helpers.py#L327